src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AlleleAndContext.java

```diff
@@ -1,19 +1,21 @@
 package org.broadinstitute.hellbender.tools.walkers.haplotypecaller;
 import htsjdk.variant.variantcontext.Allele;
+import htsjdk.variant.variantcontext.SimpleAllele;
 
 /**
  * This class is similar to {@link org.broadinstitute.hellbender.tools.walkers.haplotypecaller.LocationAndAlleles} but
  * allows keeping only an allele/ref pair rather than a list of alleles. The comparison is done on allele by allele basis and
  * not in the way it is done on LocationAndAlleles
  */
 
-public class AlleleAndContext extends Allele {
+public class AlleleAndContext extends SimpleAllele {
     final static public long serialVersionUID = 1L;
     private final int loc;
     private final String contig;
     private final Allele refAllele;
+
     public AlleleAndContext(final String contig, final int loc, final Allele allele, final Allele refAllele) {
-        super(allele, false);
+        super(allele.getBases(), allele.isReference());
```

Comment:
can we just get rid of all the old calls to this copy constructor.

Reply:
There are no other calls to it in our codebase, we can deprecate it and replace it with a more appropriately named method.

Reply:
can you also maybe depricate it because its confusing as hell and a trap?